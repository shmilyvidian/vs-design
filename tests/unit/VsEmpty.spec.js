import { shallowMount, mount } from '@vue/test-utils'
import VsEmpty from '@/packages/vs-empty/src/index.vue'

const wrapper = mount(VsEmpty)

describe('VsEmpty.vue', () => {
  // 通过快照校验组件内容变动
  test('render VsEmpty', () => {
    const wrapper = shallowMount(VsEmpty)
    expect(wrapper).toMatchSnapshot()
  })
  // 测试组件class名是否正确渲染
  it('shonld render empty-container class', () => {
    expect(wrapper.attributes().class).toMatch('vs-empty-container')
    wrapper.destroy()
  })
  // 测试props imageType是否正确渲染
  it('renders imageType to match class', () => {
    const typeArr = ['noUpdate', 'noData', 'noMessage', 'noFinish', 'noTask', 'noNetwork']
    typeArr.forEach(itemType => {
      const wrapper = shallowMount(VsEmpty, {
        propsData: { 
          imageType: itemType
        }
      })
      expect(wrapper.props().imageType).toMatch(itemType)
      wrapper.destroy()
    })
  })
  // 测试props imageSize是否正确渲染
  it('test props imageSize', () => {
    const wrapper = mount(VsEmpty, {
      propsData: {
        imageSize: '60px'
      }
    })
    expect(wrapper.props().imageSize).toContain('60px')
    wrapper.destroy()
  })
  // 测试props description是否正确渲染
  it('test empty.vue description', () => {
    const wrapper = mount(VsEmpty, {
      propsData: {
        description: 'description text'
      }
    })
    expect(wrapper.html()).toMatch('description text')
    wrapper.destroy()
  })
  // 测试props insertStyle样式是否生效
  it('test empty.vue insertStyle', () => {
    const wrapper = mount(VsEmpty, {
      propsData: {
        insertStyle: {
          description: {
            fontSize: '18px',
            paddingTop: '10px'
          }
        }
      }
    })
    const styleResult = wrapper.find('.vs-description').attributes().style
    expect(styleResult).toMatch('font-size: 18px')
    expect(styleResult).toMatch('padding-top: 10px')
    wrapper.destroy()
  })
  // 测试slots插槽传入内容是否生效
  it('test empty.vue slots', () => {
    const wrapper = mount(VsEmpty, {
      slots: {
        default: 'slot text'
      }
    })
    expect(wrapper.html()).toMatch('slot text')
    wrapper.destroy()
  })
})
