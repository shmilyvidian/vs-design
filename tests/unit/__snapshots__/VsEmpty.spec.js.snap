// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VsEmpty.vue render VsEmpty 1`] = `
<div class="vs-empty-container"><img src="" style="width: 138px;">
  <div class="vs-description"></div>
</div>
`;
